Services/DataBaseService/Repositories/UserRepository.cs

```diff
@@ -22,17 +25,44 @@ public void CreateUser(User user)
         {
             dbContext.Users.Add(mapper.MapToDbUser(user));
             dbContext.SaveChanges();
+
         }
 
         public void CreateUserCredential(UserCredential userCredential)
         {
-            dbContext.UsersCredentials.Add(mapper.MapToDbUserCredential(userCredential));
-            dbContext.SaveChanges();
+            var dbCredential = dbContext.UsersCredentials.FirstOrDefault(uc => uc.Email == userCredential.Email);
+
+            if (dbCredential is null)
+            {
+                dbContext.UsersCredentials.Add(mapper.MapToDbUserCredential(userCredential));
+                dbContext.SaveChanges();
+            }
+            else
+            {
+                throw new ForbiddenException("User already exists");
+            }
+            
         }
 
         public UserCredential GetUserCredential(string email)
         {
             var dbCredential = dbContext.UsersCredentials.FirstOrDefault(uc => uc.Email == email);

```

Comment:
May be we may ignore capital letters comparing emails?

Reply:
done